otherlibs/unix/sockopt.c

```diff
@@ -177,6 +177,9 @@ unix_getsockopt_aux(char * name,
                     enum option_type ty, int level, int option,
                     value socket)
 {
+  CAMLparam0();
+  CAMLlocal1(err);
+  value res;
```

Comment:
Shouldn't `res` be registered in `CAMLlocal`?

Reply:
Unless I'm missing something, `res` is always computed last in the function (there is no allocation after it is defined), so the current code is fine. We need `err` to be registered because there is an allocation after its definition in the TYPE_UNIX_ERROR case -- which originally had the `Begin_roots` stuff.

Reply:
(Note for other people: "fine" modulo the missing `break` in the switch of course, which were added in a follow-up fix PR.)

Reply:
I'll trust you on this, my previous source was the manual.
> Local variables of type `value` must be declared with one of the `CAMLlocal` macros.

Reply:
I distinctly remember trying to convince you to work on an OCaml/Rust FFI project, which would have taught you all about root registration :-)